src/FrameReflower/Text.php

```diff
@@ -221,7 +222,15 @@ protected function _layout_line()
         switch ($style->white_space) {
             default:
             case "normal":
-                $frame->set_text($text = $this->_collapse_white_space($text));
+                $text = $this->_collapse_white_space($text);
+
+                if ($style->direction === 'rtl') {
+                    $bidi = new Bidi($text, null, null, 'R', true);
```

Comment:
If the document says it wants RTL it should get that don't you think so?
Which problems do you want to solve with letting tc-lib-unciode deciding what to do?

If we are doing the step to pass all text it is another story surely. But I don't know if browsers are having an auto detection. I doubt that as there is just the CSS value ltr or rtl.

Reply:
I think we can keep this initial release focused on the text itself. We can focus a future release on layout modification for rtl content. Conceptually this seems straightforward, just position from the right side of the container instead of the left. I think probably what will make it difficult is locating all the places where we need to add logic for this.